src/security/srv_acl.c

```diff
@@ -506,16 +506,65 @@ get_sec_capas_for_token(Auth__Token *token, struct ownership *ownership,
 	return rc;
 }
 
+static int
+get_sec_origin_for_token(Auth__Token *token, char **machine)
+{
+	struct drpc_alloc	alloc = PROTO_ALLOCATOR_INIT(alloc);
+	int 			machine_size = 0;
+	int 			rc = 0;
+	char 			*mtmp;
+	Auth__Sys		*authsys;
+
+
+	if (token == NULL || machine == NULL) {
+		D_ERROR("NULL input\n");
+		return -DER_INVAL;
+	}
+
+	rc = get_auth_sys_payload(token, &authsys);
+	if (rc != 0)
+		return rc;
+
+	if(authsys->machinename) {
+		 machine_size = strnlen(authsys->machinename, MAXHOSTNAMELEN);
+	} else {
+		D_ERROR("Malformed AuthSys token missing machinename");
+		rc = -DER_INVAL;
+		goto out;
```

Comment:
Do we really want to fail out here if we're missing this? I guess the question is how compatible we want to be with components of older versions at this point.

Reply:
I believe we need to. The rest of the code is expecting pool handles to contain the machine they were opened on and without it being in the credential there is no way for us to get this information.